lib/benchpark/experiment.py

```diff
@@ -70,12 +70,43 @@ def compute_modifiers_section(self):
 
     def compute_applications_section(self):
         # TODO: is there some reasonable default?
+        variables = {}
+        variables["n_ranks"] = num_procs  # TODO: num_procs will be defined in the child...
+        n_resources = num_procs
+
         raise NotImplementedError(
             "Each experiment must implement compute_applications_section"
         )
 
     def compute_spack_section(self):
         # TODO: is there some reasonable default based on known variable names?
+        system_specs = {}
+        system_specs["compiler"] = "default-compiler"
+        system_specs["mpi"] = "default-mpi"
+        system_specs["lapack"] = "lapack"   #TODO: can we define this in lapack experiment?
+
+        package_specs = {}
+        package_specs[system_specs["mpi"]] = (
+            {}
+        )  # empty package_specs value implies external package
```

Comment:
This seems like a good candidate for a method in `Experiment` like

```
def needs_external(pkgs_dict, system_specs, pkg_name):
  pkgs_dict[system_specs[pkg_name]] = {}
```

(mainly so this comment doesn't have to be repeated).